staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -397,3 +397,19 @@ func (u *Unstructured) GetClusterName() string {
 func (u *Unstructured) SetClusterName(clusterName string) {
 	u.setNestedField(clusterName, "metadata", "clusterName")
 }
+
+func (u *Unstructured) GetObservedGeneration() int64 {
+	val, found, err := NestedInt64(u.Object, "status", "observedGeneration")
```

Comment:
I don't think this belongs as a first-class method... if custom resources want to do this, they can call `NestedInt64(u.Object, "status", "observedGeneration")`